tests/crypto/read_file.c

```diff
@@ -8,7 +8,12 @@
 oe_result_t read_cert(char* filename, char* cert)
 {
     size_t len_cert;
+#ifdef _WIN32
```

Comment:
To Simon's point, I would prefer not seeing ifdefs scattered around every file and since fopen_s is part of the standard (see https://en.cppreference.com/w/c/io/fopen) we should just use fopen_s.
For any platform that doesn't support it, we should have in a .h file a macro that maps it to fopen.

Reply:
It's not standard in C99, and both enclave and host use this. I'll try to integrate the ifdef _WIN32 stuff into the associated header file, readfile.h.

Reply:
Hmmm. From our discussion on https://github.com/openenclave/openenclave/issues/2376 it looks like we decided that samples and public headers should be C99 compliant. @dthaler thoughts on test code as well?

But to Ryan's point, inside the enclave _s functions are not supported and this test runs on both host and enclave, so in this particular instance we can't use just `fopen_s`

Reply:
Why is fopen_s not supported in the enclave?  Let's fix that.

Personally I don't think test code needs to be C99 compliant since test code shouldn't appear in arbitrary apps compiled with arbitrary compilers (that's what sample code is for).

Reply:
> Why is fopen_s not supported in the enclave? Let's fix that.

No _s functions are supported. I think it was a specific design decision, but I wasn't around at the time so I can really speak to the reasoning/point to a gh issue. I have heard something along the lines of that these APIs are mostly regarded in the Linux community as not very useful.

Is your thought that all the secure C11 functions should be available, or just specific ones?

> Personally I don't think test code needs to be C99 compliant since test code shouldn't appear in arbitrary apps compiled with arbitrary compilers (that's what sample code is for).

I agree. Thanks for the input.

Reply:
@jhand2 @dthaler I've changed the other things but this one proves to take more time than I'd thought. They take different arguments in different formats so I will need to write another wrapper to handle `fopen()`/`fopen_s()`, as well as `sscanf()/sscanf_s()` near line 125. The compiler complained even though I rewrote it using `__STDC_VERSION__` macros.
and the compiler still found a way to complain about all the `fopen_s()` and `sscanf_s()`.

On the other hand, I've revised the other issues accordingly, and it surely looks neat after all the changes. Are there any other concerns?

Reply:
@jhand2 I think all the secure standard C11 functions should be available in an enclave.  One has to be able to write apsp that compile with a C99 compliant compiler, but that doesn't mean C11 functions cannot be available.